### PR TITLE
Create infrastructure for SemanticDB

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
I plan to refactor common logic with the decompiler later on.

File: compiler/src/dotty/tools/dotc/semanticdb/TASTyConsumer.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package dotty.tools.dotc.semanticdb
+
+import dotty.tools.dotc.core.Contexts._
+import dotty.tools.dotc.core.Phases.Phase
+import dotty.tools.dotc.tastyreflect.TastyImpl
+
+class TASTyConsumer extends Phase {
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/semanticdb/TASTyConsumer.scala]: Tasty not TASTy

File: compiler/src/dotty/tools/dotc/semanticdb/Main.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+object Main extends dotc.Driver {
+  override protected def newCompiler(implicit ctx: Context): dotc.Compiler = {
+    assert(ctx.settings.fromTasty.value)
+    new TASTyFromClass
+  }
+
+  override def setup(args0: Array[String], rootCtx: Context): (List[String], Context) = {
+    var args = args0.filter(a => a != "-gen-semantic-db")
+    if (!args.contains("-from-tasty")) args = "-from-tasty" +: args
+    if (args.contains("-d")) args = "-color:never" +: args
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/semanticdb/Main.scala]: ?

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/semanticdb/Main.scala]: Copy pasted from the decompiler. Will remove it.

**Comment by allanrenucci** [comment]
What about a compiler plugin instead?

File: project/Build.scala [hunk 3]
```diff
@@ -677,6 +677,7 @@ object Build {
     val dottyCompiler = jars("dotty-compiler")
     val args0: List[String] = spaceDelimited("<arg>").parsed.toList
     val decompile = args0.contains("-decompile")
+    val consumeTasty = args0.contains("-gen-semantic-db")
```
Comment (olafurpg) [inline project/Build.scala]: nit: I'd write `-gen-semanticdb` and in in camel case it would be `GenSemanticdb` (or `SemanticdbGen`, ..)

Comment (nicolasstucki) [reply project/Build.scala]: Actually I should remove this flag.

**Comment by nicolasstucki** [comment]
I agree with the names. They are horrible. Any suggestions?

**Comment by smarter** [comment]
no idea

**Comment by nicolasstucki** [comment]
I will ask around to see if someone has a good idea

**Pull request merged**